src/test/scala/gcd/GcdTesters2.scala

```diff
@@ -0,0 +1,53 @@
+// See README.md for license details.
+
+package gcd
+
+import chisel3._
+import chisel3.tester._
+import org.scalatest.FreeSpec
+import chisel3.experimental.BundleLiterals._
+
+/**
+  * This is a trivial example of how to run this Specification
+  * From within sbt use:
+  * {{{
+  * testOnly gcd.GcdDecoupledTester
+  * }}}
+  * From a terminal shell use:
+  * {{{
+  * sbt 'testOnly gcd.GcdDecoupledTester'
+  * }}}
+  */
+//scalastyle:off magic.number
+class GcdTesters2 extends FreeSpec with ChiselScalatestTester {
+
+  "Gcd should calculate proper greatest common denominator" in {
+    test(new DecoupledGcd(16)) { dut =>
+      dut.input.initSource()
+      dut.input.setSourceClock(dut.clock)
+      dut.output.initSink()
+      dut.output.setSinkClock(dut.clock)
+
+      val testValues = for { x <- 0 to 10; y <- 0 to 10} yield (x, y)
+      val inputSeq = testValues.map { case (x, y) => (new GcdInputBundle(16)).Lit(_.value1 -> x.U, _.value2 -> y.U) }
+      val resultSeq = testValues.map { case (x, y) =>
+        (new GcdOutputBundle(16)).Lit(_.value1 -> x.U, _.value2 -> y.U, _.gcd -> BigInt(x).gcd(BigInt(y)).U)
+      }
+
+      fork {
```

Comment:
I'd make orthogonal tests of correctness and stall-tolerance behavior, instead of combining them into one big test. Because small targeted tests tell you exactly what is wrong, while massive tests only tell you something is wrong and leaves it to you to figure out what.

Reply:
I agree with this. Looking at this test now, it uses chiseltest, but it's very much like an iotesters test where everything is crammed into one test. It would seem reasonable to work this into a sequence of tests like what you'd expect from scalatest.